error.go

```diff
@@ -199,6 +199,12 @@ func wrapAll(errs ...error) error {
 }
 
 func isTLSError(err error) bool {
+	var errEDB Error
+
+	if errors.As(err, &errEDB) {
+		err = errEDB.Unwrap()
```

Comment:
It seems like it might be more robust to use `errors.As()` with the x509 error types this function checks for. That way if the error chain changes again in the future this wont break.